src/cardstack/al-cardstack-view.ts

```diff
@@ -285,41 +323,82 @@ export abstract class AlCardstackView< EntityType=any,
         if ( filterProperties.length === 0 ) {
-            return card;
+            return true;
         }
-        filterProperties.find( property => {
+        filterProperties.forEach( property => {
             if ( ! card.properties.hasOwnProperty( property ) || typeof( ( card.properties as any)[property] ) === 'undefined' ) {
-                card.visible = false;
-                return true;        //  terminate iteration
+                visible = false;
+                return ;        //  terminate iteration
             }
             let cardPropValue = ( card.properties as any )[property];
             let matched = Object.values( this.activeFilters[property] ).find( valDescriptor => {
-                return valDescriptor.value === cardPropValue ? true : false;
+                if (cardPropValue instanceof Array) {
+                    return cardPropValue.includes(valDescriptor.value);
+                }
+                return valDescriptor.value === cardPropValue.value;
             } );
             if ( ! matched ) {
-                card.visible = false;
-                return true;        //  terminate iteration
+                visible = false;
+                return ;        //  terminate iteration
             }
-            return false;
+            return ;
         } );
-        return card;
+        return visible;
+    }
+
+    protected evaluateCardVisibilityBySearch( card:AlCardstackItem<EntityType,PropertyType>, search: RegExp|null):boolean {
+        let visible = false;
+        if (search === null) {
+            return true;
+        }
+
+        if(this.characteristics && this.characteristics.searchableBy) {
+            if (this.characteristics.searchableBy.length === 0 ) {
+                return true;
+            }
+            this.characteristics.searchableBy.forEach( (property:string) => {
+                if ( ! card.properties.hasOwnProperty( property ) || typeof( ( card.properties as any)[property] ) === 'undefined' ) {
+                    return ; //  terminate iteration
+                }
+                let cardPropValue = ( card.properties as any )[property];
+                if (cardPropValue instanceof Array) {
+                    const matches = cardPropValue.find((value) => search.test(value));
+                    if (matches) {
+                        visible = true;
+                        return;
+                    }
+                } else {
+                    if (search.test(cardPropValue)) {
+                        visible = true;
+                    }
+                }
+                return ;
+            });
+        }
+        return visible;
+    }
+
+    protected evaluateCardState( card:AlCardstackItem<EntityType,PropertyType> ):AlCardstackItem<EntityType,PropertyType> {
+            card.visible = false;
+            // filter using state active or inactive
+            if (this.evaluateCardVisibilityBySearch(card, this.textFilter)
```

Comment:
I would simplified this into the simple expression

`card.visible = this.evaluateCardVisibilityByFilter(card) && this.evaluateCardVisibilityBySearch( card, this.textFilter ) )`.